src/mame/includes/fantland.h

```diff
@@ -53,40 +51,36 @@ class fantland_state : public driver_device
 	required_device<screen_device> m_screen;
 	required_device<palette_device> m_palette;
 	required_device<generic_latch_8_device> m_soundlatch;
-	optional_device<msm5205_device> m_msm1;
-	optional_device<msm5205_device> m_msm2;
-	optional_device<msm5205_device> m_msm3;
-	optional_device<msm5205_device> m_msm4;
+	optional_device_array<msm5205_device, 4> m_msm;
+	optional_region_ptr<uint8_t> m_adpcm_rom;
```

Comment:
Why not make a derived state class for the systems that have the MSM5205 chips?  Then you can use required device/region finders here and get validation benefits.